packages/core/tests/controllers/SwapController.test.ts

```diff
@@ -1,78 +1,72 @@
+// MATIC
+const networkToken = tokenInfo[0] as SwapTokenWithBalance
+// FOX
 const toToken = tokenInfo[1] as SwapTokenWithBalance
 
 // - Helpers
-function setSourceTokenAmount(value: string) {
-  SwapController.setSourceTokenAmount(value)
-  const toTokenAmount = SwapController.getToAmount()
-  const toTokenValues = SwapController.getToTokenValues(toTokenAmount, toToken?.decimals)
-  SwapController.state.toTokenAmount = toTokenValues.toTokenAmount
-  SwapController.state.toTokenPriceInUSD = toTokenValues.toTokenPriceInUSD
+function initializeSwapState(_sourceToken: SwapTokenWithBalance, _toToken: SwapTokenWithBalance) {
+  SwapController.state.tokensPriceMap = prices
+  SwapController.state.networkPrice = prices[networkTokenAddress].toString()
+  SwapController.state.networkBalanceInUSD = networkToken.quantity.numeric
+  SwapController.state.gasPriceInUSD = SwapCalculationUtil.getGasPriceInUSD(
+    SwapController.state.networkPrice,
+    gasLimit,
+    gasFee
+  )
+  SwapController.setSourceToken(_sourceToken)
+  SwapController.state.sourceTokenPriceInUSD = _sourceToken.price
+  SwapController.setToToken(_toToken)
+  SwapController.state.toTokenPriceInUSD = _toToken.price
+  SwapController.state.toTokenAmount = SwapCalculationUtil.getToTokenAmount({
+    sourceToken: _sourceToken,
+    sourceTokenAmount: '1',
+    sourceTokenPrice: _sourceToken.price,
+    toToken: _toToken,
+    toTokenPrice: _toToken.price
+  })
 }
 
 // - Setup ---------------------------------------------------------------------
 beforeAll(() => {
   AccountController.setCaipAddress(caipAddress)
-  SwapController.state.tokensPriceMap = prices
-  SwapController.state.networkPrice = prices[sourceTokenAddress].toString()
-  SwapController.state.networkBalanceInUSD = '2'
-  SwapController.state.gasPriceInUSD = SwapController.calculateGasPriceInUSD(gasLimit, gasFee)
-  SwapController.setSourceToken(sourceToken)
-  SwapController.state.sourceTokenPriceInUSD = sourceToken.price
-  SwapController.setToToken(toToken)
-  setSourceTokenAmount('1')
+})
+
+beforeEach(() => {
+  initializeSwapState(networkToken, toToken)
 })
 
 // -- Tests --------------------------------------------------------------------
 describe('SwapController', () => {
-  it('should set toToken as expected', () => {
-    expect(SwapController.state.toToken?.address).toEqual(toToken.address)
-  })
-
   it('should set sourceToken as expected', () => {
-    expect(SwapController.state.sourceToken?.address).toEqual(sourceToken.address)
-  })
-
-  it('should calculate gas price in Ether and USD as expected', () => {
-    const gasPriceInEther = SwapController.calculateGasPriceInEther(gasLimit, gasFee)
-    const gasPriceInUSD = SwapController.calculateGasPriceInUSD(gasLimit, gasFee)
-
-    expect(gasPriceInEther).toEqual(0.068395033074)
-    expect(gasPriceInUSD).toEqual(0.06395499714651795)
+    expect(SwapController.state.sourceToken?.address).toEqual(networkToken.address)
```

Comment:
shouldn't you test the usage of the method here? 
feels weird to rely on the beforeAllHook for this 🤔

Reply:
This makes sure that when user opens the Swap page, the first token should be selected. The `beforeAll` sounds not completely reliable but we have to be sure that the first token is selected when the Swap is ready.

Maybe we should call `initializeSwapState` in every test?